lesson08/main/Dockerfile

```diff
@@ -0,0 +1,17 @@
+FROM golang:1.16-alpine
```

Comment:
This is an ok way to build an image, but in production this is usually done in a better way. I know that hw description did not specify that and it was not covered in the lecture, so this will suffice, and you don't need to change it. I just give it to you as general information.
This docker image will be quite heavy, since it contains all the go environment, as well as your source files, and there are three ways to make it lighter:
1. a tedious way. Clean up, remove everything that is not needed to run the result binary(go folder, source files, evrything else, that was used only for compilation, and not during runtime)
2. a better way. You build your binary locally on your maching, but you need to keep in mind, that you need to build it for the OS, on which this binary will be run. So if the result OS is alpine, you need to build it for alpine. After that you pack the image using clean alpine base image.
3. probably best and cleanest way for this case. Use multi-stage build, where your Dockerfile contains two stages, practically creating two images - on the first stage the binary is built, and on the second this same binary is packed into the base alpine image. You can read about multi-stage builds [here](https://docs.docker.com/develop/develop-images/multistage-build/)

Reply:
ok, i also noticed that image was quite heavy. Will keep that in mind.